shell/components/ResourceDetail/index.vue

```diff
@@ -1,569 +1,70 @@
-     * information.
-     * @param {string|null} resourceOverride - An optional resource override
-     * string. If not provided, the method will use the default resource from
-     * the route parameters or the instance's resourceOverride property.
-     */
-    configureResource(userId = '', resourceOverride = null) {
-      const id = userId || this.$route.params.id;
-      const resource = resourceOverride || this.resourceOverride || this.$route.params.resource;
-      const options = this.$store.getters[`type-map/optionsFor`](resource);
-
-      const detailResource = options.resourceDetail || options.resource || resource;
-      const editResource = options.resourceEdit || options.resource || resource;
-
-      // FIXME: These aren't right... signature is (rawType, subType).. not (rawType, resourceId)
-      // Remove id? How does subtype get in (cluster/node)
-      this.detailComponent = this.$store.getters['type-map/importDetail'](detailResource, id);
-      this.editComponent = this.$store.getters['type-map/importEdit'](editResource, id);
-    },
-    /**
-     * Sets the mode and initializes the resource components.
-     *
-     * This method sets the mode of the component and configures the resource
-     * components based on the provided user and resource.
-     *
-     * @param {Object} payload - An object containing the mode, user, and
-     * resource properties.
-     * @param {string} payload.mode - The mode to set.
-     * @param {Object} payload.user - The user object containing user-specific
-     * information.
-     * @param {string} payload.resource - The resource string to use for
-     * initialization.
-     */
-    setMode({ mode, userId, resource }) {
-      this.mode = mode;
-      this.value.id = userId;
-      this.configureResource(userId, resource);
-    }
-  }
+// Ideally I'd prefer to have separate routes/pages for each of these but our app makes a
+// fair amount of assumptions around having one detail page for each resource and that the
+// detail, config, edit, yaml, create pages are all derived from the same page.
+//
+// I could also dynamically check for and import these pages but I wanted this to be easier
+// to be explicit and easier to search for.
+const resourceToPage: any = {
+  // 'apps.daemonset':    require('@shell/pages/explorer/resource/detail/apps.daemonset.vue').default,
+  // 'apps.deployment':   require('@shell/pages/explorer/resource/detail/apps.deployment.vue').default,
+  // 'apps.statefulset':  require('@shell/pages/explorer/resource/detail/apps.statefulset.vue').default,
+  // 'batch.cronjob':     require('@shell/pages/explorer/resource/detail/batch.cronjob.vue').default,
+  // 'batch.job':         require('@shell/pages/explorer/resource/detail/batch.job.vue').default,
+  // 'cluster-dashboard': require('@shell/pages/explorer/resource/detail/cluster-dashboard.vue').default,
+  configmap: require('@shell/pages/explorer/resource/detail/configmap.vue').default,
```

Comment:
optional: Consider if using dynamic imports instead of `require` can fit for this use case. Previous investigations have shown that `require` behaves in such a way that webpack is unable to split code dynamically, leading to larger bundle sizes.

#12246